### DISCUSSION TITLE
Automatic generation Sequence diagrams from dynamic diagrams

### DISCUSSION THREAD

**Discussion by dawidci-papaya** [avisi-cloud/structurizr-site-generatr#351] [Q&A] [answered]
Hi, 

Is it possible or do You have any plans for automatic generation sequence diagrams from dynamic diagrams ?

Structurizr dynamic diagrams are good but i prefer to visualize processes in Sequence diagram. 

For now i can simply use ChatGPT to parse dynamic diagram to sequence diagram but this is still manual work. It will be great to have such mechanizm directly in tool.

**Comment by jp7677** [comment]
To be honest, not really. Also I'm not sure if such a diagram transformation should even be part of this project. May be https://github.com/avisi-cloud/structurizr-site-generatr/pull/338 can help with your case.

**Comment by qtzar** [comment]
I noticed that the latest release of the Structurizr-Export library now has some support for rendering as Sequence Diagrams. Maybe once that dependency gets merged in I can take a look at it and see how it could be implemented.

**Comment by dirkgroot** [comment, marked as answer]
I'm not sure how long this has been available already, but there is a view property available called `plantuml.sequenceDiagram`. This property can be set to `true` for a single diagram or for all diagrams, to generate sequence diagrams instead of collaboration diagrams.

See https://docs.structurizr.com/export/plantuml for more information.

**Comment by dawidci-papaya** [comment]
@dirkgroot  This is exactly what I want to achieve.
Thanks For help !
